src/internal/m365/graph/uploadsession.go

```diff
@@ -75,5 +84,21 @@ func (iw *largeItemWriter) Write(p []byte) (int, error) {
 	// Update last offset
 	iw.lastWrittenOffset = endOffset
 
+	// Once the upload is complete, we get a Location header in the
+	// below format from which we can get the id of the uploaded
+	// item. This will only be available after we have uploaded the
+	// entire content(based on the size in the req header).
+	// https://outlook.office.com/api/v2.0/Users('<user-id>')/Messages('<message-id>')/Attachments('<attachment-id>')
```

Comment:
is there some Microsoft documentation that explains this? It would give some added comfort as presumably it would be some spec they'll stick to going forward

Reply:
oh, also is this code shared by events and messages? Does that change anything if it is?

Reply:
https://learn.microsoft.com/en-us/graph/outlook-large-attachments?tabs=http explains this. I'll add the link to comments. Also, it is same for both mail and event.